xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/resources/xwiki-resource/plugin.js

```diff
@@ -265,8 +265,17 @@ require(['jquery', 'resource', 'resourcePicker'], function ($, $resource) {
         if (onlyTheReferencePart  === true) {
           this.setValue(resourceReference.reference);
         } else {
-          this.setValue(CKEDITOR.plugins.xwikiResource.getResourceURL(resourceReference,
-            this.getDialog().getParentEditor()));
+          var resourceURL = CKEDITOR.plugins.xwikiResource.getResourceURL(resourceReference,
+            this.getDialog().getParentEditor());
+          var width = this.getDialog().getValueOf(resourcePickerId[0], "width");
+          if (width) {
+            resourceURL = resourceURL + '&width=' + width;
+          }
+          var height = this.getDialog().getValueOf(resourcePickerId[0], "height");
+          if (height) {
+            resourceURL = resourceURL + '&height=' + height;
+          }
```

Comment:
This is very hackish. width/height don't make sense for the generic resource plugin. Not all resources have an width/height.